### DISCUSSION TITLE
Streamflow simulation giving nan values for initial 90 days

### DISCUSSION THREAD

**Discussion by Meelisha** [neuralhydrology/neuralhydrology#74] [Q&A] [answered]
Hi all!

Really having a great time exploring the package. This has been extremely helpful. I was looking for some insights on something that I encountered. 

So, I have trained an EA-LSTM model with 531 basins that are in the CAMELS dataset. I have used 29 years (1985-2014) of data to train the model for 30 epochs. The median NSE value was around 0.7. Using this model, I tried simulating streamflow for 11 out-of-sample basins and it gave very low NSE values (below zero). 
So, I then finetuned another EA-LSTM model initializing with weights from the above model. Since, all the basins are not in the CAMELS dataset, generic dataset was used. This training is done for 100 epochs and using 29 years of data from the generic dataset created above. This resulted in pretty good NSE values with few exceptions. However, looking at the simulation of the streamflow from the finetuned model, I see that around 90 days of the simulation have nan values. This also happens for pre-trained model (531 basins trained) simulations of the said basins. I am not sure how this is happening. I checked for input values that might have nan values, but, I don't find any.

Can anyone suggest a direction I should look towards to get to the root of this? 

Thanks!

**Comment by Meelisha** [comment, marked as answer]
The thing I missed out in the above case was I had input data from 1980-01-01 and I was trying to simulate streamflow starting from 1980-10-01 and I had used 365 days as the sequence length. Since, there were not enough input data (365) for October 1st 1980 to December 31st 1980, I was seeing the nan values. A basic but fundamental idea was missed here.

**Comment by kratzert** [comment]
Sorry for not replying earlier but great to see that you found the problem
